Jagati.cmake

```diff
@@ -1991,13 +1991,21 @@ macro(EmitExceptionSource)
     set(NamespaceSourceHeader "\
 #include \"DataTypes.h\"\n\
 #include \"MezzException.h\"\n\n\
+SAVE_WARNING_STATE\n\n\
 namespace Mezzanine\n{\n\
 namespace Exception\n{\n\
 \n"
     )
 
     set(NamespaceHeader "\
 #include \"DataTypes.h\"\n\n\
+SAVE_WARNING_STATE\n\
+SUPPRESS_CLANG_WARNING(\"-Winconsistent-missing-destructor-override\")\n\
```

Comment:
I am assuming this line suppresses a warning that the destructor (that destroys some object....) is not correct? Maybe the destructor destroys JagatiVersion?

Reply:
Take a look in your build directory look for `config/MezzException.h`. This Jagati code generates this file. 

In the Foundation CMakeLists.txt look around line 100 for a bunch of AddJagatiException that inform what goes into this file. When using many Jagati packages all of the exceptions will be assembled togther into one file. This allows the Jagati to create some exception meta that is intended to facilitate troubleshooting in the future.

In Java exceptions include a ton of metadata, like a stack trace and a linked list of other exceptions. The detault C++ includes a single pointer to a string, and it might be null.

We made our own set of exceptions optimized for games. A big linked list and stack tracing are likely to impact performance. So we exclude those, but a unique error number, line number, file name, and function names are cheap to track and very useful in troubleshooting. It is the middle ground that is ideal for our purposes. This is part of the system that generates this information.